pkg/workload/schemachange/operation_generator.go

```diff
@@ -307,6 +323,16 @@ func (og *operationGenerator) addColumn(ctx context.Context, tx pgx.Tx) (*opStmt
 			code: pgcode.FeatureNotSupported, condition: isJSONTyp,
 		},
 	})
+	// Compatibility errors aren't guaranteed since the cluster version update is not
+	// fully transaction aware.
+	op.potentialExecErrors.addAll(codesWithConditions{
+		{code: pgcode.Syntax, condition: pgLSNNotSupported},
```

Comment:
are duplicate codes handled here? it might be better to make one variable like:
```
	isNotFinalized, err := isClusterVersionLessThan(
		ctx,
		tx,
		clusterversion.ByKey(clusterversion.V23_2))
```

also, we can add one more check that we actually are using these types, so we end up with:
```
	op.potentialExecErrors.addAll(codesWithConditions{
		{code: pgcode.Syntax, condition: isNotFinalized && (isPGLSN || isRefcursor)},
Comment
		{code: pgcode.FeatureNotSupported, condition: isNotFinalized && (isPGLSN || isRefcursor)},
		{code: pgcode.UndefinedObject, condition: isNotFinalized && (isPGLSN || isRefcursor)},
	})
```

(take a look at how `isJSONTyp` is defined above.)